projects/graph/graph.py

```diff
@@ -7,42 +7,112 @@ class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
         self.vertices = {}
+        self.queue = Queue()
+        self.stack = Stack()
     def add_vertex(self, vertex):
         """
         Add a vertex to the graph.
         """
-        pass  # TODO
+        self.vertices[vertex] = set()
     def add_edge(self, v1, v2):
         """
         Add a directed edge to the graph.
         """
-        pass  # TODO
+        if v1 not in self.vertices:
+            self.add_vertex(v1)
+        else:
+            self.vertices[v1].add(v2)
+
     def bft(self, starting_vertex):
         """
         Print each vertex in breadth-first order
         beginning from starting_vertex.
         """
-        pass  # TODO
+
+        visited = []
+        vertices = self.vertices
+        queue = self.queue
+        queue.enqueue(starting_vertex)
+        while queue.size():
+            cur_ver = queue.dequeue()
+            if cur_ver not in visited:
```

Comment:
A very solid implementation here Lorenzo 👍 All I would do is to add in a print statement before you the `cur_ver` to `visited` so you can check out the traversal in the terminal.